src/compiler/parser.ts

```diff
@@ -1124,6 +1124,14 @@ namespace ts {
             return token === t && tryParse(nextTokenCanFollowModifier);
         }
 
+        function nextTokenIsOnSameLineAndCanFollowModifier() {

```

Comment:
So we have nextTokenIsOnSameLineAndCanFollowModifier and nextTokenCanFollowModifier...

These are supremely easy to confuse.  When would i want one, versus the other?